File: lab4/Figures/Domain/Cylinder/Cylinder.cpp
#include "Cylinder.hpp"
#include "../../../../Utils/StringUtils.hpp"

double Cylinder::GetHeight() const
{
    return height;
}

Cylinder::operator std::string() const
{
    return StringUtils::StringConcatenator() << density <<  ' ' << volume << ' ' << GetMass() << ' ' << height;
}

File: lab4/Bodies/lib/Figures/Cylinder/Cylinder.dart
import '../BodyRotation/BodyRotation.dart';

class Cylinder extends BodyRotation
{
  double _height;

  Cylinder(double density, double volume, double radius, double this._height): super(density, volume, radius) {}

  double GetHeight() => _height;

  @override
  String toString() {
    return 'Cylinder: ' + 'V = ' + GetVolume().toString() + ' R = ' + GetBaseRadius().toString() +
    ' Dens = ' + GetDensity().toString() + ' H = ' + _height.toString();
  }
}
